main.go

```diff
@@ -48,6 +49,7 @@ func main() {
 	flagset.StringVar(&unsafePassthroughPaths, "unsafe-passthrough-paths", "", "Comma delimited allow list of exact HTTP path segments should be allowed to hit upstream URL without any enforcement."+
 		"This option is checked after Prometheus APIs, you can cannot override enforced API to be not enforced with this option. Use carefully as it can easily cause a data leak if the provided path is an important"+
 		"API like targets or configuration. NOTE: \"all\" matching paths like \"/\" or \"\" and regex are not allowed.")
+	flagset.BoolVar(&errorOnReplace, "error-on-replace", false, "When specified proxy will return 403 if the query already contains a label matcher that differs from the on the proxy would inject.")
```

Comment:
nit: I think we can make this a full sentence:
```suggestion
	flagset.BoolVar(&errorOnReplace, "error-on-replace", false, "When specified, the proxy will return HTTP status code 403 if the query already contains a label matcher that differs from the on the proxy would inject.")
```